template/README.md

```diff
@@ -366,7 +367,75 @@ module.name_mapper='^\(.*\)\.\(jpg\|png\|gif\|eot\|svg\|ttf\|woff\|woff2\|mp4\|w
 
 We will consider integrating more tightly with Flow in the future so that you don’t have to do this.
 
-### Deploy
+### Adding Custom Environment Variables
+
+> Note: this feature is available with `react-scripts@0.3.0` and higher.
+
+Your project can consume variables declared in your environment as if they were declared locally in your JS files. By
+default you will have `NODE_ENV` defined for you, and any other environment variables starting with
+`REACT_APP_`. These environment variables will be defined for you on `process.env`. For example, having an environment
+variable named `REACT_APP_SECRET_CODE` will be exposed in your JS as `process.env.REACT_APP_SECRET_CODE`, in addition
+to `process.env.NODE_ENV`.
+
+These environment variables can be useful for displaying information conditionally based on where the project is
+deployed or consuming sensitive data that lives outside of version control.
+
+First, you need to have environment variables defined, which can vary between OSes. For example, let's say you wanted to
+consume a secret defined in the environment inside a `<form>`:
+
+```jsx
+<h6>Hello, Admin!<h6>
+
+<small>You are running this application in <strong>{process.env.NODE_ENV}</strong> mode.</small>
+
+<form>
+  <input type="hidden" defaultValue={process.env.REACT_APP_SECRET_CODE} />
+</form>
+```
+
+The above form is looking for a variable called `REACT_APP_SECRET_CODE` from the environment. In order to consume this
+value, we need to have it defined in the environment:
+
+```bash
+# Create an environment variable on Windows:
+set REACT_APP_SECRET_CODE=abcdef
+
+# Create an environment variable on macOS/Linux/Unix
+REACT_APP_SECRET_CODE=abcdef
```

Comment:
I probably won't use the words "Bash" since bash is becoming a thing on Windows. But yes, I'll separate the blocks.

@alexzherdev yes, just like @gaearon's snippet between us, it doesn't use export to set the variable. Export is used for permanence of variables, but as I mentioned that's outside the scope of this.